source/server/http/admin_filter.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <functional>
+#include <list>
+
+#include "envoy/http/filter.h"
+#include "envoy/server/admin.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/logger.h"
+#include "common/http/codes.h"
+#include "common/http/header_map_impl.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Server {
+
+/**
+ * A terminal HTTP filter that implements server admin functionality.
+ */
+class AdminFilter : public Http::StreamDecoderFilter,
+                    public AdminStream,
+                    Logger::Loggable<Logger::Id::admin> {
+public:
+  AdminFilter(std::function<Http::Code(absl::string_view path_and_query,
+                                       Http::ResponseHeaderMap& response_headers,
+                                       Buffer::OwnedImpl& response, AdminFilter& filter)>
+                  admin_server_run_callback_func);
+
+  // Http::StreamFilterBase
+  void onDestroy() override;
+
+  // Http::StreamDecoderFilter
+  Http::FilterHeadersStatus decodeHeaders(Http::RequestHeaderMap& headers,
+                                          bool end_stream) override;
+  Http::FilterDataStatus decodeData(Buffer::Instance& data, bool end_stream) override;
+  Http::FilterTrailersStatus decodeTrailers(Http::RequestTrailerMap& trailers) override;
+  void setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callbacks) override {
+    callbacks_ = &callbacks;
+  }
+
+  // AdminStream
+  void setEndStreamOnComplete(bool end_stream) override { end_stream_on_complete_ = end_stream; }
+  void addOnDestroyCallback(std::function<void()> cb) override;
+  Http::StreamDecoderFilterCallbacks& getDecoderFilterCallbacks() const override;
+  const Buffer::Instance* getRequestBody() const override;
+  const Http::RequestHeaderMap& getRequestHeaders() const override;
+
+private:
+  typedef std::function<Http::Code(absl::string_view path_and_query,
```

Comment:
use `using` rather than `typedef` here.

Also can you put this in the `public` section and re-use it? There's also a repeat of this typedef in tests.